### ISSUE TITLE
Allow use of h/j/k/l for movement within XIKI

### ISSUE THREAD

**Issue by alexlafroscia** [trogdoro/xiki#139]
I'm not sure how it would work, considering Xiki tries to use key presses to filter visible results, but it would be really useful to be able to use `h/j/k/l` to navigate a la `vim`.  As a Vim user, I've trained myself to avoid the arrow keys, and needing to switch back to them for Xiki is a pain!

**Comment by trogdoro** [comment]
Note that you can use ^B, ^F, ^P, and ^N to move back, forward, previous
line, and next line.

--Craig

**Comment by aemonge** [comment]
Well yeah, but that emacs style tough =/

**Comment by trogdoro** [comment]
I'm prepending "Feature request: " to this.

Or, does anyone have better ideas about how to distinguish between feature requests and issues on github?

**Comment by alexlafroscia** [comment]
Labels? I find that they're pretty easy, since you can search/sort by them and you get some nice color-coding going on

**Comment by trogdoro** [comment]
@alexlafroscia Great idea. I changed the name back and added a "Feature Request" label.

**Comment by KolesnichenkoDS** [comment]
> use h/j/k/l to navigate

Great idea

> I'm not sure how it would work, considering Xiki tries to use key presses to filter visible results

Just typing **/word** or **:word** instead of **word** can be ok

**Comment by trogdoro** [comment]
> use h/j/k/l to navigate

With the next version of Xiki (not yet pushed) you'll be able to hold
down alt (or option) and type h,j,k or l to navigate like in vim.

--Craig

**Comment by alexlafroscia** [comment]
Would it be possible to use `CTL-H/J/K/L` instead? I have `CAPSLOCK` mapped to it for comfort and try to stay away for ALT since it's harder to reach.

**Comment by trogdoro** [comment]
> Would it be possible to use CTL-H/J/K/L instead?

That would step on a lot of existing keys. I could probably make a key
shortcut that temporarily enabled CTL-H/J/K/L, or just H/J/K/L if there was
enough interest from vim people.

> I have CAPSLOCK mapped to it for comfort

Me too :)

> try to stay away for ALT since it's harder to reach.

I also have Tab mapped to alt (when used like a modifier) via Karabiner (a
mac utility).

--Craig

**Comment by alexlafroscia** [comment]
> I also have Tab mapped to alt (when used like a modifier) via Karabiner (a
mac utility).

Oh wow, that's interesting.  I use Karabiner and Seil to make tapping `CAPSLOCK` mirror `ESC` and holding mirror `CTRL`, I'll have to check out remapping `TAB` as well.

**Comment by trogdoro** [comment]
>  I use Karabiner and Seil to make tapping CAPSLOCK mirror ESC

That sounds kind of cool - I may try that. It seems strange that
Karabiner can't do that on its own since Seil is made by the same
developer. But it looks like everyone is saying you need both
(Karabiner and Seil) to do that >
http://www.economyofeffort.com/2014/08/11/beyond-ctrl-remap-make-that-caps-lock-key-useful/

I currently type Ctrl-[ for ESC. On most terminals they're equivalent.
The next version of Xsh will make much more extensive use of the ESC
key for canceling out of things and escaping out of views.

--Craig

**Comment by trogdoro** [comment]
Actually the 3rd screenshot shows a way to map all control keys (not
just caps lock) to ESC when you tap them using just Karabiner (and the
built-in osx control to capslock feature):

http://www.economyofeffort.com/2014/08/11/beyond-ctrl-remap-make-that-caps-lock-key-useful/

Thanks for the tip - will try out tapping caps lock now instead of
Ctrl-[ ! All vim users should know about that option - could save some
serious carpal tunnel :)

--Craig

**Comment by trogdoro** [comment]
>> make tapping CAPSLOCK mirror ESC
> That sounds kind of cool - I may try that

It's been 2 days and I'm loving tapping CAPSLOCK for ESC. Thanks again
for the tip!!

Any other related mappings or productivity shortcuts I should know about? :)

--Craig

**Comment by alexlafroscia** [comment]
No, that's really the big one that I use.  It took a little time to re-train my hands to not hit the `ESC` key but now that I'm used to it, I've realized how much of a time-saver it is!

I guess maybe my only other suggestion is using `SPACE` as my leader key, which I find really useful because it puts it under both thumbs, and your hands are already right over it.
